hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -2893,7 +2890,12 @@ private AbfsLease maybeCreateLease(String relativePath, TracingContext tracingCo
     if (!enableInfiniteLease) {
       return null;
     }
-    AbfsLease lease = new AbfsLease(client, relativePath, tracingContext);
+    final AbfsLease lease;
+    if (getPrefixMode() == PrefixMode.DFS) {
```

Comment:
Should we have a check here that the lease config is enabled ?

Reply:
That lease config `fs.azure.lease.create.non.recursive` is specially for having createNonRecursive add. It is added because in lease is something new which we have added in createNonRecursive.
Here lease take is already there in the trunk / ABFS_3.3.2_dev branch.